src/test/java/org/springframework/integration/aws/inbound/SnsInboundChannelAdapterTests.java

```diff
@@ -133,7 +133,7 @@ public void testUnsubscribe() throws Exception {
 		this.mockMvc.perform(
 				post("/mySampleTopic")
 						.header("x-amz-sns-message-type", "UnsubscribeConfirmation")
-						.contentType(MediaType.APPLICATION_JSON)
+						.contentType(MediaType.TEXT_PLAIN)
```

Comment:
Just leave one of them as `MediaType.APPLICATION_JSON` and push.
Will merge than immediately and you'll be able to test the solution in your application against the latest BUILD-SNAPSHOT.

I will be also appreciate if you can link this our issue into that Amazon one. To refresh their heard and let them know that people still make bump on their forehead with the problem and waste the time for workaround it. :cry: